internal/controller/siteconfig_controller.go

```diff
@@ -191,40 +205,332 @@ func (r *SiteConfigReconciler) handleFinalizer(ctx context.Context, siteConfig *
+		msg := fmt.Sprintf("failed to validate rendered manifests for SiteConfig %s using dry-run validation", siteConfig.Name)
+		if err != nil {
+			msg = fmt.Sprintf(", err: %v", err)
+		}
+		r.Log.Info(msg)
+
+		conditions.SetStatusCondition(&siteConfig.Status.Conditions,
+			conditions.RenderedTemplatesValidated,
+			conditions.Failed,
+			metav1.ConditionFalse,
+			"Rendered manifests failed dry-run validation")
+	} else {
+		conditions.SetStatusCondition(&siteConfig.Status.Conditions,
+			conditions.RenderedTemplatesValidated,
+			conditions.Completed,
+			metav1.ConditionTrue,
+			"Rendered templates validation succeeded")
+	}
+
+	if updateErr := conditions.PatchStatus(ctx, r.Client, siteConfig, patch); updateErr != nil {
+		if err == nil {
+			r.Log.Info(fmt.Sprintf("failed to update SiteConfig %s status post validation of rendered templates, err: %s", siteConfig.Name, updateErr.Error()))
+			err = updateErr
+		}
+	}
+	if !rendered || err != nil {
+		return
+	}
+
+	// Apply the rendered manifests
+	r.Log.Info(fmt.Sprintf("Applying rendered manifests for SiteConfig %s", siteConfig.Name))
+	patch = client.MergeFrom(siteConfig.DeepCopy())
+	rendered, err = r.executeRenderedManifests(ctx, r.Client, siteConfig, manifestGroups, v1alpha1.ManifestRenderedSuccess)
+	if err != nil || !rendered {
+		msg := fmt.Sprintf("failed to apply rendered manifests for SiteConfig %s", siteConfig.Name)
+		if err != nil {
+			msg = fmt.Sprintf(", err: %v", err)
+		}
+		r.Log.Info(msg)
+
+		conditions.SetStatusCondition(&siteConfig.Status.Conditions,
+			conditions.RenderedTemplatesApplied,
+			conditions.Failed,
+			metav1.ConditionFalse,
+			"Failed to apply site config manifests")
+	} else {
+		conditions.SetStatusCondition(&siteConfig.Status.Conditions,
+			conditions.RenderedTemplatesApplied,
+			conditions.Completed,
+			metav1.ConditionTrue,
+			"Applied site config manifests")
+	}
+
+	if updateErr := conditions.PatchStatus(ctx, r.Client, siteConfig, patch); updateErr != nil {
+		if err == nil {
+			r.Log.Info(fmt.Sprintf("Failed to update SiteConfig %s status post creation of rendered templates, err: %s", siteConfig.Name, updateErr.Error()))
+			err = updateErr
+		}
+	}
+	return
+}
+
+func (r *SiteConfigReconciler) handleSuppressedManifests(ctx context.Context, siteConfig *v1alpha1.SiteConfig) error {
+	patch := client.MergeFrom(siteConfig.DeepCopy())
+
+	suppressFn := func(suppressedManifests []string) {
+		for _, kind := range suppressedManifests {
+			for index, manifest := range siteConfig.Status.ManifestsRendered {
+				if manifest.Kind == kind {
+					siteConfig.Status.ManifestsRendered[index].Status = v1alpha1.ManifestSuppressed
+					siteConfig.Status.ManifestsRendered[index].Message = ""
+				}
+			}
+		}
+	}
+
+	// Suppress cluster-level manifests
+	suppressFn(siteConfig.Spec.SuppressedManifests)
+
+	// Suppress node-level manifests
+	for _, node := range siteConfig.Spec.Nodes {
+		suppressFn(node.SuppressedManifests)
+	}
```

Comment:
We allow this today?

Reply:
Yes, currently to support the deletion of BMH hosts, we have to suppress generating the BMH manifest (ref: https://docs.openshift.com/container-platform/4.15/edge_computing/ztp-advanced-install-ztp.html#ztp-deleting-node-siteconfig_ztp-advanced-install-ztp).

Reply:
Seems strange but OK.
I'd expect the API will be to delete the node from the site config.

Reply:
This is a long-standing issue with SiteConfig that came up when we added the "remove node from spoke cluster" feature.

If you delete the node from the SiteConfig then the NMStateConfig also gets deleted. This is required if we want to do cleaning on the node after it's removed from the spoke cluster. To get around the issue SiteConfig users have to suppress the node manifests, wait for the host to be deprovisioned, then delete them.

It's a bit of an architectural issue since we group NMState with infraenv (which is a cluster resource for SiteConfig) and SiteConfig groups NMState with the node.